sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -146,7 +146,14 @@ def __init__(self, packages, options, environment_version):
     if self.standard_options.streaming:
       job_type = 'FNAPI_STREAMING'
     else:
-      job_type = 'PYTHON_BATCH'
+      use_fnapi_for_batch = (
+          self.debug_options.experiments
+          and 'beam_fn_api' in self.debug_options.experiments
+      )
+      if use_fnapi_for_batch:
+        job_type = 'FNAPI_BATCH'
```

Comment:
Batch and streaming use the same container, the logic is already in dependency.py: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/dataflow/internal/dependency.py#L501, https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/dataflow/internal/dependency.py#L533.